policies/branch-policy.md

```diff
@@ -0,0 +1,100 @@
+  deficiency is present. It can be then merged (backported or directly
+  cherry-picked) to all older branches where the deficiency is present.
+
+## A feature branch
+
+- There exists one major feature branch where collective development of
+  a major feature X is happening.
+- Any changes that are needed only for the X feature are merged to this branch.
+- To ensure easy mergeability of the feature branch to master once the
+  development is finished, the branch is periodically (once a month) rebased
+  on top of the master branch. The rebase might happen more often if there is
+  already a larger amount of changes accumulated in the master branch.
+- The review process for the feature branch is the same as the review process
+  for the master branch except for the rebases where only casual review is
+  expected but the CI must pass fully before the force-push of the rebased
+  branch is allowed.
+- Once the development of the feature X is completed the OTC approves merge
+  of the freshly rebased feature branch to the master branch.
+
+## A next minor branch
+
+- The development of the next minor release happens there.
+- No API/ABI breaking changes are allowed.
+- No major features are allowed unless explicitly acked by OMC as targeted for
+  the minor release.
+- No major refactoring is allowed.
+- Any changes (features, bug-fixes, documentation, ...) done on the next-minor
+  branch must be forward-ported or directly cherry-picked to the master branch
+  if applicable.
+- Exceptions are possible for example when we are removing deprecated
+  functionality in the master branch.
+- Backporting of features and refactoring changes from the master branch to the
+  next-minor branch happens on a case-by-case basis decided by OTC. **Can we
+  find criteria for making the process less heavyweight on OTC and more
+  expectable by outside contributors? For example we might say that minor
+  features/refactorings are implicitly allowed but it is up to the PR submitter
+  to say whether he wants the PR to be merged to both master and next-minor
+  or just master.**
+
+Branch and tag naming policy
+----------------------------
+
+The branch where the development of the next major release is happening is
+called `master`.
+
+**In case the next-minor policy is not chosen to be approved:** The branch where
+the development of the next minor release is happening is the same as the branch
+above.
+
+The existing stable release branches are named `openssl-x.y`.
+
+As a legacy exception to the rule above, the branch where the development of
+OpenSSL-1.1.1 fix releases is happening is called `OpenSSL_1_1_1-stable`.
+
+Feature branch: The branch where the feature XYZ is developed is called
+`feature-xyz`.
+
+Next-minor: The branch where the development of the next minor release is
+happening is called `next-minor`. After the beta release of the minor version
+the branch is renamed to `openssl-x.y`.
```

Comment:
I am not sure that actual branch renaming will interact nicely with git clients.
Wouldn't the second time we use `next-minor` incur a non-fast-forward update to the branch?

Reply:
Yeah it would I suppose. In that case we might name the branch openssl-x.y from start. I just wanted to give an option to the OMC to decide to use some different y version although that seems not too important. Also renaming (or simply re-branchng) openssl-3.1 to openssl-3.13 if OMC decided that the next version will have that minor number seems less problematic.

Reply:
I'll modify the proposal.